cg/meta/upload/scout/uploadscoutapi.py

```diff
@@ -191,77 +191,46 @@ def upload_fusion_report_to_scout(
         scout_api: ScoutAPI = self.scout
         status_db: Store = self.status_db
         report_type: str = "Research" if research else "Clinical"
+        rna_case: models.Family = status_db.family(case_id)
 
-        fusion_report: Optional[hk_models.File] = self.get_fusion_report(case_id, research)
-        if fusion_report is None:
-            raise FileNotFoundError(
-                f"{report_type} fusion report was not found in housekeeper for {case_id}"
-            )
+        map_dict = self.create_rna_dna_mapper(rna_case=rna_case)
+        for rna_key in map_dict.keys():
+            fusion_report: Optional[hk_models.File] = self.get_fusion_report(case_id, research)
+            if fusion_report is None:
+                raise FileNotFoundError(
+                    f"{report_type} fusion report was not found in housekeeper for {case_id}"
+                )
 
-        LOG.info(f"{report_type} fusion report %s found", fusion_report.path)
+            LOG.info(f"{report_type} fusion report %s found", fusion_report.path)
 
-        rna_case = status_db.family(case_id)
         upload_dna_cases: Set[models.Family] = set()
         for link in rna_case.links:
             rna_sample: models.Sample = link.sample
             if not rna_sample.subject_id:
                 raise CgDataError(
-                    "Failed on RNA sample %s since it is not linked to anything via subject_id"
-                    % rna_sample.internal_id,
+                    "Failed on RNA sample %s as subject_id field is empty" % rna_sample.internal_id,
                 )
 
-            dna_cases: [models.Family] = self.status_db.families_by_subject_id(
-                customer_id=rna_sample.customer.internal_id,
-                data_analyses=[Pipeline.MIP_DNA, Pipeline.BALSAMIC],
-                subject_id=rna_sample.subject_id,
-                is_tumour=rna_sample.is_tumour,
-            )
-
-            self._check_cases_single_connected(dna_cases, rna_sample)
+            for rna_key, dna_value in map_dict.items():
+                for dna_sample_name, dna_cases in dna_value.items():
+                    for dna_case_id in dna_cases:
```

Comment:
This should work and remove two for loops 👍 
```suggestion
            dna_sample_id, case_list = map_dict[rna_sample.internal_id].pop_item()
                    for dna_case_id in case_list:
```

Reply:
Great, thanks! It will be part of the next commit.